src/main/java/org/team1540/robot2025/FieldConstants.java

```diff
@@ -0,0 +1,157 @@
+                                                    .getX(),
+                                            poseDirection
+                                                    .transformBy(new Transform2d(adjustX, adjustY, new Rotation2d()))
+                                                    .getY(),
+                                            level.height),
+                                    new Rotation3d(
+                                            0,
+                                            Units.degreesToRadians(level.pitch),
+                                            poseDirection.getRotation().getRadians())));
+                    fillLeft.put(
+                            level,
+                            new Pose3d(
+                                    new Translation3d(
+                                            poseDirection
+                                                    .transformBy(new Transform2d(adjustX, -adjustY, new Rotation2d()))
+                                                    .getX(),
+                                            poseDirection
+                                                    .transformBy(new Transform2d(adjustX, -adjustY, new Rotation2d()))
+                                                    .getY(),
+                                            level.height),
+                                    new Rotation3d(
+                                            0,
+                                            Units.degreesToRadians(level.pitch),
+                                            poseDirection.getRotation().getRadians())));
+                }
+                branchPositions.add(fillRight);
+                branchPositions.add(fillLeft);
+            }
+        }
+    }
+
+    public static class StagingPositions {
+        // Measured from the center of the ice cream
+        public static final Pose2d leftIceCream =
+                new Pose2d(Units.inchesToMeters(48), Units.inchesToMeters(230.5), new Rotation2d());
+        public static final Pose2d middleIceCream =
+                new Pose2d(Units.inchesToMeters(48), Units.inchesToMeters(158.5), new Rotation2d());
+        public static final Pose2d rightIceCream =
+                new Pose2d(Units.inchesToMeters(48), Units.inchesToMeters(86.5), new Rotation2d());
+    }
+
+    public enum ReefHeight {
+        L1(Units.inchesToMeters(25.0), 0),
+        L2(Units.inchesToMeters(31.875), -35),
+        L3(Units.inchesToMeters(47.625), -35),
+        L4(Units.inchesToMeters(72), -90);
+
+        ReefHeight(double height, double pitch) {
+            this.height = height;
+            this.pitch = pitch; // in degrees
+        }
+
+        public static ReefHeight fromLevel(int level) {
+            return Arrays.stream(values())
+                    .filter(height -> height.ordinal() == level)
+                    .findFirst()
+                    .orElse(L4);
```

Comment:
seriously just make an array

Reply:
or a switch case

Reply:
Actually just

```suggestion
        public static ReefHeight fromLevel(int level) {
            return values()[level-1]
```

Reply:
Bro i stole this code check the file header. But ok i will look into making it less painful for you

Reply:
just because you stole it doesn't make it good

Reply:
yeah ik that's why i will fix it